app/models/search/query.rb

```diff
@@ -113,19 +113,20 @@ def generated_query
       from: pagination_offset,
       sort: sort
     }
-    if aggregations.present?
-      q.merge!(aggregations)
+    if (aggs = aggregations).present?
+      q.merge!(aggs)
     end
     q
   end
 
-  # Combine the filters and queries
+  # Combine the filters and queries, with a fallback in case there are no
```

Comment:
I added a pair of tests checking for `match_all` on the `PseudQuery` and `TagQuery` (and also reorganized those tests to avoid having to delete/rebuild indexes on the tests that didn't need it, so you might want to ignore whitespace on the diff). That should also cover the new branch in `make_bool`, since that's why I had to introduce the `|| { match_all: {} }` code in the first place.

IMO, the [sorting tests](https://github.com/otwcode/otwarchive/blob/38f7d81af0ef83363f62d77d5919edf0c66deb39/spec/models/search/bookmark_search_form_spec.rb#L15-L134) in `spec/models/search/bookmark_search_form_spec.rb` are pretty good coverage for the changes to the way that `field_value_score` works. They're testing behavior rather than implementation, which is why I didn't have to modify them in this PR — this PR isn't supposed to change the surface behavior at all.

I also threw in some extra tests to make sure the facet behavior hasn't changed.